lib/puppet/node.rb

```diff
@@ -23,8 +23,8 @@ class Puppet::Node
 
   def initialize_from_hash(data)
     @name = data['name']
-    @classes = data['classes']
-    @parameters = data['parameters']
+    @classes = data['classes'] || []
+    @parameters = data['parameters'] || {}
     @environment_name = data['environment']
```

Comment:
environment_name can be nil, if so, then it looks up the environment lazily. In practice 'name' can not be nil since the normal initializer requires a name. You would have to have a broken serialization for that to happen. It seems prudent to check name, and fail the same way as the normal initializer. I will add that.